include/swift/SIL/SILInstruction.h

```diff
@@ -2416,7 +2416,9 @@ class FunctionRefInst : public FunctionRefBaseInst {
   ///
   /// \param DebugLoc  The location of the reference.
   /// \param F         The function being referenced.
-  FunctionRefInst(SILDebugLocation DebugLoc, SILFunction *F);
+  /// \param context   The type expansion context of the function reference.
+  FunctionRefInst(SILDebugLocation DebugLoc, SILFunction *F,
+                  TypeExpansionContext context);
```

Comment:
For instructions and basic block arguments, could they get their TypeExpansionContext from the containing `SILFunction`?

Reply:
SILInstructions don't have access to their parent basic block (or SILFunction) on creation. The TypeExpansionContext is needed at creation to compute the SILType of the instruction.

Reply:
Makes sense, I see that the `SILBuilder` APIs haven't changed and use the builder's current context too. Thanks!